web-ui/src/main/java/org/archcnl/ui/input/ConceptAndRelationView.java

```diff
@@ -13,22 +19,68 @@
     CreateNewLayout createNewRelationLayout =
             new CreateNewLayout("Relations", "Create new relation");
     HorizontalLayout bottomBarLayout = new HorizontalLayout();
-
-    public void setUpBottomBar() {
-        final Button saveButton = new Button("Save");
-        final Button checkButton = new Button("Check");
-        bottomBarLayout.add(saveButton, checkButton);
-    }
+    TreeGrid<ListEntry<Concept>> conceptTreeGrid = new TreeGrid<>();
+    TreeGrid<ListEntry<Relation>> relationTreeGrid = new TreeGrid<>();
 
     public ConceptAndRelationView() {
         createNewConceptLayout.setHeight(50, Unit.PERCENTAGE);
         createNewRelationLayout.setHeight(50, Unit.PERCENTAGE);
 
         setUpBottomBar();
+        setUpConceptView();
+        setUpRelationView();
 
         add(createNewConceptLayout);
         add(createNewRelationLayout);
         add(bottomBarLayout);
         getStyle().set("border", "1px solid black");
+
+        createNewConceptLayout.add(conceptTreeGrid);
+        createNewRelationLayout.add(relationTreeGrid);
+    }
+
+    public void setUpBottomBar() {
+        final Button saveButton = new Button("Save");
+        final Button checkButton = new Button("Check");
+        bottomBarLayout.add(saveButton, checkButton);
+    }
+
+    public void setUpConceptView() {
```

Comment:
Debatable but maybe it makes sense to export the list views for Concept and Relation into its own class. This way the definition of the nearly identical methods setUpConceptView and setUpRelation View could be avoided. Also, this would later allow for easy online switching of the specific view implementation to let's say a tile view...

Reply:
Yes, you are right, so I extracted the behavior into its own class. As I forgot to mention anywhere else, I did not implement the tile view yet, as I think it would be easier if everything regarding the hierarchy and maybe other changes are done beforehand, so that they do not have to be adapted to the List and the Tile view.